pkg/infrastructure/data_index.go

```diff
@@ -78,3 +87,87 @@ func InjectDataIndexURLIntoSupportingService(client *client.Client, namespace st
 	}
 	return nil
 }
+
+// MountProtoBufConfigMapsOnDeployment mounts protobuf configMaps from KogitoRuntime services into the given deployment
+func MountProtoBufConfigMapsOnDeployment(client *client.Client, deployment *appsv1.Deployment) (err error) {
+	cms, err := getProtoBufConfigMapsForAllRuntimeServices(deployment.Namespace, client)
+	if err != nil || len(cms.Items) == 0 {
+		return err
+	}
+	for _, cm := range cms.Items {
+		appendProtoBufVolumeIntoDeployment(deployment, cm)
+		appendProtoBufVolumeMountIntoDeployment(deployment, cm)
+	}
+	updateProtoBufPropInToDeploymentEnv(deployment)
+	return nil
+}
+
+// MountProtoBufConfigMapOnDataIndex mounts protobuf configMaps from KogitoRuntime services into the given deployment instance of DataIndex
+func MountProtoBufConfigMapOnDataIndex(client *client.Client, kogitoService v1alpha1.KogitoService) (err error) {
+	deployment, err := getSupportingServiceDeployment(kogitoService.GetNamespace(), client, v1alpha1.DataIndex)
+	if err != nil || deployment == nil {
+		return
+	}
+
+	cms, err := getProtoBufConfigMapsForSpecificRuntimeService(client, kogitoService.GetName(), kogitoService.GetNamespace())
+	if err != nil || len(cms.Items) == 0 {
+		return
+	}
+	for _, cm := range cms.Items {
+		appendProtoBufVolumeIntoDeployment(deployment, cm)
+		appendProtoBufVolumeMountIntoDeployment(deployment, cm)
+	}
+	updateProtoBufPropInToDeploymentEnv(deployment)
+	return kubernetes.ResourceC(client).Update(deployment)
+}
+
+func appendProtoBufVolumeIntoDeployment(deployment *appsv1.Deployment, cm corev1.ConfigMap) {
+	for _, volume := range deployment.Spec.Template.Spec.Volumes {
+		if volume.Name == cm.Name {
+			return
+		}
+	}
+
+	// append volume if its not exists
+	deployment.Spec.Template.Spec.Volumes =
+		append(deployment.Spec.Template.Spec.Volumes, corev1.Volume{
+			Name: cm.Name,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					DefaultMode: &framework.ModeForProtoBufConfigMapVolume,
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: cm.Name,
+					},
+				},
+			},
+		})
+}
+
+func appendProtoBufVolumeMountIntoDeployment(deployment *appsv1.Deployment, cm corev1.ConfigMap) {
+	for fileName := range cm.Data {
+		mountPath := path.Join(defaultProtobufMountPath, cm.Labels["app"], fileName)
+		for _, volumeMount := range deployment.Spec.Template.Spec.Containers[0].VolumeMounts {
+			if volumeMount.MountPath == mountPath {
+				return
+			}
+		}
+
+		// append volume mount if its not exists
+		deployment.Spec.Template.Spec.Containers[0].VolumeMounts =
+			append(deployment.Spec.Template.Spec.Containers[0].VolumeMounts, corev1.VolumeMount{
+				Name:      cm.Name,
```

Comment:
Minor? In case config map contains more protobuf files the Name parameter here is duplicated for every volume mount entry for those protobuf files. Shouldn't it be rather unique? Though I haven't see any issue with it so far.